types/react-dom/canary.d.ts

```diff
@@ -91,7 +91,7 @@ declare module "." {
         as: PreinitAs;
         crossOrigin?: "anonymous" | "use-credentials" | "" | undefined;
         fetchPriority?: "high" | "low" | "auto" | undefined;
-        precedence?: string | undefined;
+        precedence?: "reset" | "low" | "medium" | "high" | undefined;
```

Comment:
This is not how `precedence` works. You probably confuse this with "priority" as a concept. React allows arbitrary strings here because the app developer decides which precedences there are.

The React docs just need to be updated i.e. apply https://github.com/reactjs/react.dev/pull/6908/ to https://react.dev/reference/react-dom/preinit#parameters